lib/buffer.js

```diff
@@ -65,6 +71,9 @@ function alignPool() {
  * would ever actually be removed.
  **/
 function Buffer(arg, encodingOrOffset, length) {
+  if (!(this instanceof Buffer)) {
+    bufferDeprecationWarning();
+  }
```

Comment:
@jasnell Done. I kept the wording because "Calling Buffer() without `new` is deprecated." implies that calling Buffer _with_ `new` is _not_ deprecated.